src/features/device-state/routes/state.ts

```diff
@@ -395,3 +419,195 @@ export const state: RequestHandler = async (req, res) => {
 		res.sendStatus(500);
 	}
 };
+
+const getDevice = async (req: Request, uuid: string) => {
+	const device = await sbvrUtils.db.readTransaction!((tx) =>
+		stateQuery()({ uuid }, undefined, { req, tx }),
+	);
+
+	if (!device) {
+		throw new UnauthorizedError();
+	}
+
+	return device;
+};
+
+const getConfig = (device: any) => {
+	const config: Dictionary<string> = {};
+
+	// add any app-specific config values...
+	const userAppFromApi: AnyObject = device.belongs_to__application[0];
+	varListInsert(userAppFromApi.application_config_variable, config);
+
+	// override with device-specific values...
+	varListInsert(device.device_config_variable, config);
+	filterDeviceConfig(config, device.os_version);
+	setMinPollInterval(config);
+
+	return config;
+};
+
+const getUserAppForState = async (device: any, config: Dictionary<string>) => {
+	const userAppFromApi: AnyObject = device.belongs_to__application[0];
+
+	// get the release of the main app that this device should run...
+	const release = getReleaseForDevice(device);
+
+	// grab the main app for this device...
+	return release == null
+		? {
+				name: userAppFromApi.app_name,
+				services: {},
+				networks: {},
+				volumes: {},
+				uuid: userAppFromApi.uuid,
+				appId: userAppFromApi.id,
+		  }
+		: await buildAppFromRelease(device, userAppFromApi, release, config);
+};
+
+const getSupervisorAppForState = async (
+	req: Request,
+	device: any,
+	config: Dictionary<string>,
+) => {
+	if (device.should_be_managed_by__release.length === 0) {
+		return undefined;
+	}
+
+	const fromApi = await sbvrUtils.db.readTransaction!(async (tx) => {
+		const resinApiTx = api.resin.clone({ passthrough: { req, tx } });
+		return await resinApiTx.get({
+			resource: 'application',
+			id: device.should_be_managed_by__release[0].belongs_to__application.__id,
+			options: {
+				$select: ['id', 'app_name', 'uuid'],
+				$expand: {
+					application_config_variable: {
+						$select: ['name', 'value'],
+						$orderby: {
+							name: 'asc',
+						},
+					},
+					application_environment_variable: {
+						$select: ['name', 'value'],
+						$orderby: {
+							name: 'asc',
+						},
+					},
+				},
+			},
+		});
+	});
+
+	if (!fromApi) {
+		return undefined;
+	}
+
+	fromApi.should_be_running__release = device.should_be_managed_by__release;
+
+	return await buildAppFromRelease(
+		device,
+		fromApi,
+		device.should_be_managed_by__release[0],
+		config,
+	);
+};
+
+const getDependent = async (device: any) => {
+	const userAppFromApi: AnyObject = device.belongs_to__application[0];
+
+	const dependent = {
+		apps: {} as AnyObject,
+		devices: {} as AnyObject,
+	};
+
+	const depAppCache: Dictionary<{
+		release?: AnyObject;
+		application_environment_variable: Array<{
+			name: string;
+			value: string;
+		}>;
```

Comment:
I think this should be
```suggestion
		application_environment_variable: Array<Pick<ApplicationEnvironmentVariable, 'name' | 'value'>>;
```